### ISSUE TITLE
configurable whitespaces

### ISSUE THREAD

**Issue by michaelrsweet** [michaelrsweet/mxml#123]
Version: -current
Original reporter: 

full (long) context at http://groups.google.com/group/harbour-devel/browse_thread/thread/996b2344aaaeb1/f0b9f5da238e4b80

the rundown is: even though white spaces are not of significance in xml, we would like to ask for configurable white spaces (spaces and end-of-line markers, ie. cr, lf or crlf). compile-time configurability would be a start, but run-time configurability would be best.

it is accepted that chaning MXML_TAB and a custom whitespace_cb can take care half the story, but there are eols hardcoded to 0x0a in at least mxml-file.c, and also i seem to remember one of the save functions injecting spaces here and there, although i can't seem to find this one now.

basically we should at will like to be able to produce xml strings with no white spaces (new lines or between nodes or anything) at all, and when having a formatted output, we should like to be able to control the aspects that are indifferent for compliant parsers, but have a significance when the result is worked on by a human on various platforms.
